UserDetails/forms.py

```diff
@@ -52,12 +57,141 @@ class Meta:
+                                settings.DURATION_AFTER_MEMBERSHIP_REJECTION > timezone.now():
+                            # The user has been verified not to be a member to recently (prevent spamming)
+                            self.disabled = True
+
+            except ObjectDoesNotExist:
+                pass
+        if association is None:
+            raise ValueError("Association can not be None")
+
+    def verified(self):
+        if self.membership is None:
+            return None
+        return self.membership.get_verified_state()
+
+    def get_membership_model(self, user=None, new_value=True):
+        # Check input data for correctness
+        if self.user is None and user is None:
+            raise ValueError("Field does not contain user and user was not given in method")
+        if user is not None and self.user is not None and self.user != user:
+            raise ValueError("Given user differs from field user")
+
+        if self.membership is not None:
+            return self.membership
+        if self.user is not None:
+            # If there was a user given, but the link was not found. Create a new link if allowed
+            if new_value:
+                return UserMembership(related_user=self.user, association=self.association)
+        else:
+            # user originally not given. Try to find the link
+            try:
+                return self.association.usermembership_set.get(related_user=user)
+            except ObjectDoesNotExist:
+                if new_value:
+                    return UserMembership(related_user=user, association=self.association)
+        return None
+
+
+class AssociationLinkForm(forms.Form):
+
+    def __init__(self, user, *args, **kwargs):
+        super(AssociationLinkForm, self).__init__(*args, **kwargs)
+
+        self.user = user
+
+        if user is None:
+            associations = Association.objects.filter(is_choosable=True)
+        else:
+            associations = Association.objects.filter(
+                Q(is_choosable=True) |
+                (Q(is_choosable=False) & Q(usermembership__related_user=user))) \
+                .distinct().order_by('slug')
+
+        # Get all associations and make a checkbox field
+        for association in associations:
+            field = AssociationLinkField(user, association)
+
+            self.fields[field.label] = field
+
+    def clean(self):
+        cleaned_data = super(AssociationLinkForm, self).clean()
+        has_association = False
+        for key, value in self.cleaned_data.items():
+            if value:
+                has_association = True
+
+        if not has_association:
+            raise ValidationError("At least one association needs to be chosen")
+
+        return cleaned_data
+
+    def save(self, user=None):
+        """
+        Saves the association links. Removes
+        :return:
+        """
+        if self.user is None and user is None:
```

Comment:
```suggestion
        if not self.user and not user:
```
I'd suggest for readability to only use `is None` when it is really needed. (Just a suggestion for the future)

Reply:
Ever since the querie code failed due to an incorrect None check on a queryset check I've been more rigid in my None checks, but I'll try to balance it out more.